pkg/collector/corechecks/network-devices/cisco-sdwan/sdwan.go

```diff
@@ -0,0 +1,211 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package ciscosdwan implements NDM Cisco SD-WAN corecheck
+package ciscosdwan
+
+import (
+	"time"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/network-devices/cisco-sdwan/client"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/network-devices/cisco-sdwan/payload"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/network-devices/cisco-sdwan/report"
+	"github.com/DataDog/datadog-agent/pkg/snmp/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+const (
+	// CheckName is the name of the check
+	CheckName = "ciscosdwan"
+)
+
+// Configuration for the Cisco SD-WAN check
+type checkCfg struct {
+	Hostname    string `yaml:"hostname"`
+	Port        int    `yaml:"port"`
+	Username    string `yaml:"username"`
+	Password    string `yaml:"password"`
+	Namespace   string `yaml:"namespace"`
+	MaxAttempts int    `yaml:"max_attempts"`
+	MaxPages    int    `yaml:"max_pages"`
+	MaxCount    int    `yaml:"max_count"`
+	Lookback    int    `yaml:"lookback"`
```

Comment:
I'm not sure what this is, is there maybe a better name for it?

Reply:
It's the lookback time window we use for querying metrics, e.g. if set to 30 mins, we query metrics timeseries between `now` and `now - 30mins`
Is `LookbackTimeWindow` clearer?

Reply:
`LookbackTimeWindowMinutes`